src/spyglass/lfp/v1/lfp_artifact.py

```diff
@@ -0,0 +1,366 @@
+import warnings
+from functools import reduce
+from typing import Union
```

Comment:
Are there features of the codebase that would prevent upgrading python versions for typing improvements? I can do a deeper dive on this when I tackle conda installation times

Reply:
I can't remember if it is datajoint specifically or some other package but we are definitely limited right now.